client/client.go

```diff
@@ -2,21 +2,141 @@ package client
 
 import (
 	"context"
+	"net/http"
 
 	"github.com/coreos/etcd-operator/pkg/spec"
+
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/runtime/schema"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
 )
 
 // Operator operators etcd clusters atop Kubernetes.
 type Operator interface {
-	// Create creates an etcd cluster with the given spec.
+	// Create creates an etcd cluster.
 	Create(ctx context.Context, name string, spec *spec.ClusterSpec) error
 
-	// Delete deletes the etcd cluster with given name
+	// Delete deletes the etcd cluster.
 	Delete(ctx context.Context, name string) error
 
 	// Update updates the etcd cluster with the given spec.
 	Update(ctx context.Context, name string, spec *spec.ClusterSpec) error
 
-	// Status gets the status of the etcd cluster with the given name.
-	Status(ctx context.Context, name string) (error, *spec.ClusterStatus)
+	// Get gets the etcd cluster information.
+	Get(ctx context.Context, name string) (*spec.EtcdCluster, error)
+}
+
+type operator struct {
+	tprClient *rest.RESTClient
+	tprName   string
```

Comment:
i changed it to use the const. but we limit users to use the fixed tpr name for no good reason rather than initial simplicity. and we will probably make it configurable soon. so users can have multiple operators under the same namespace.